OwnList/OwnList/OwnList.cs

```diff
@@ -0,0 +1,294 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace OwnList
+{
+    internal class OwnList<T> : IOwnList<T>
```

Comment:
Based on the requirements, we should be able to use your custom list in the foreach loop, how come your collection doesn't implement IEnumerable?

Reply:
Because of duck typing we can avoid implementing of this interface, we can just write method GetEnumerator and it will work in the same way. Am i right ?